src/hotspot/os/linux/compilerThreadTimeout_linux.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "compiler/compilerThread.hpp"
+#include "runtime/osThread.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+#include "compilerThreadTimeout_linux.hpp"
+#include "signals_posix.hpp"
+
+#include <pthread.h>
+
+#ifndef PRODUCT
+#ifdef ASSERT
+void compiler_signal_handler(int signo, siginfo_t* info, void* context) {
+  CompilerThread::current()->timeout()->compiler_signal_handler(signo, info, context);
+}
+
+void CompilerThreadTimeoutLinux::compiler_signal_handler(int signo, siginfo_t* info, void* context) {
+  switch (signo) {
+    case TIMEOUT_SIGNAL: {
+      assert(false, "compile task timed out");
```

Comment:
Can you somehow also print the task which caused the timeout? Will just accessing `CompilerThread::current()->task()` work?